aikau/src/main/resources/alfresco/layout/AlfStackContainer.js

```diff
@@ -0,0 +1,558 @@
+       * @default null
+       */
+      paneSelectionTopic: null,
+
+      /**
+       * The topic whose publication should trigger the addition of a pane
+       * 
+       * @instance
+       * @type {string}
+       * @default null
+       */
+      paneAdditionTopic: null,
+
+      /**
+       * The topic whose publication should trigger the deletion of a pane
+       * 
+       * @instance
+       * @type {string}
+       * @default null
+       */
+      paneDeletionTopic: null,
+
+      /**
+       * If this optional value is provided, a change to the hash of this variable will update the pane selection
+       *
+       * @instance
+       * @type {string}
+       * @default null
+       */
+      paneSelectionHashVar: null,
+      
+      /**
+       * Variable to record the title of the selected pane at initialisation
+       *
+       * @instance
+       * @type {object}
+       * @default null
+       * @private
+       */
+      _defaultPaneTitle: null,
+
+      /**
+       * @instance
+       */
+      postCreate: function alfresco_layout_AlfStackContainer__postCreate() {
+
+         // Initialise a StackContainer instance and watch its selectedChildWidget event
+         this.stackContainerWidget = new StackContainer({
+            style: "height: " + this.height + "; width: " + this.width + ";",
+            doLayout: this.doLayout
+         }, this.containerNode);
+         this.stackContainerWidget.watch("selectedChildWidget", lang.hitch(this, "_paneChanged"));
+
+         // Setup child widgets and startup()
+         if (this.widgets)
+         {
+            // By default we want to ensure that we don't unnecessarily process widgets for 
+            // panes that are not immediately visible. Therefore unless specifically requested 
+            // in the configuration to be the selected pane or to render immediately then
+            // we'll mark them all as delayed processing. If no pane is marked as selected then
+            // we'll ensure that the first pane is both selected and will be immediately rendered
+            var paneSelected = false;
+            array.forEach(this.widgets, function(widget) {
+               if (widget.delayProcessing !== false && !widget.selected)
+               {
+                  widget.delayProcessing = true;
+               }
+               paneSelected = paneSelected || widget.selected;
+            });
+            if (this.widgets.length && !paneSelected)
+            {
+               this.widgets[0].selected = true;
+               this.widgets[0].delayProcessing = false;
+            }
+
+            // Now add panes for each widget...
+            array.forEach(this.widgets, lang.hitch(this, "addWidget"));
+         }
+         this.stackContainerWidget.startup();
+
+         // Subscribe to some optional topics
+         if (this.paneSelectionTopic != null)
+         {
+            this.alfSubscribe(this.paneSelectionTopic, lang.hitch(this, "paneSelect"));
```

Comment:
Better to reference the function directly (i.e. pass through a pointer, rather than an object and property name)

e.g. `lang.hitch(this, this.paneSelect)`